src/Screen.js

```diff
@@ -177,6 +187,17 @@ class Navigator {
     }
   }
 
+  _checkLastAction(params) {
+    if (Date.now() - this._lastAction.timestamp < 800
```

Comment:
How did you choose this value? What if the pushed screen takes longer to render?

Semi related: is there a real scenario where pushing the same screen multiple times was actually desired? I mean, would it break something if you omitted the timestamp altogether?